File: targets/watch-tests/Fixtures.swift
import Foundation
@testable import KinetiqWatchCore

enum Fixtures {
    /// The bundled `bounds.json`, read from the target folder as the app reads it from its bundle.
    static let bounds: Bounds = {
        let url = URL(fileURLWithPath: #filePath)
            .deletingLastPathComponent()
            .deletingLastPathComponent()
            .appendingPathComponent("watch/bounds.json")
        return (try? Data(contentsOf: url)).flatMap(Bounds.decode) ?? .fallback
    }()

    static func item(_ overrides: [String: Any] = [:]) -> [String: Any] {
        var item: [String: Any] = [
            "id": "rit_1",
            "exerciseId": "wger:73",
            "exerciseName": "Bench Press",
            "sets": 4,
            "reps": "8-10",
            "weightKg": 60,
            "restSeconds": 120,
            "notes": NSNull()
        ]
        item.merge(overrides) { _, new in new }
        return item
    }

    static func snapshot(
        items: [[String: Any]] = [item()],
        routines count: Int = 1,
        overrides: [String: Any] = [:]
    ) -> Data {
        var document: [String: Any] = [
            "format": "kinetiq.watch-routines",
            "version": 1,
            "exportedAt": "2026-09-25T10:00:00.000Z",
            "unitSystem": "metric",
            "routines": (0..<count).map { index in
                ["id": "rtn_\(index)", "name": "Push", "items": items] as [String: Any]
            }
        ]
        document.merge(overrides) { _, new in new }
        return (try? JSONSerialization.data(withJSONObject: document)) ?? Data()
    }

    static func temporaryStore() -> FileStore {
        FileStore(directory: FileManager.default.temporaryDirectory
            .appendingPathComponent("kinetiq-tests-\(UUID().uuidString)", isDirectory: true))
    }
}

File: src/watch/__tests__/fixtures.ts
import type { WatchInboxEntry } from '../../../modules/watch-bridge';
import type { WatchWorkoutDocument } from '../format';

export const UUID = '7A1D0C3E-1111-4222-8333-944455556666';

/** A valid document; overrides may break it on purpose, but only under the format's own keys. */
export function document(
  overrides: Partial<Record<keyof WatchWorkoutDocument, unknown>> = {},
): Record<string, unknown> {
  const valid: WatchWorkoutDocument = {
    format: 'kinetiq.watch-workout',
    version: 1,
    id: UUID,
    routineId: 'rtn_1',
    title: 'Push',
    startedAt: '2026-09-25T10:00:00.000Z',
    endedAt: '2026-09-25T10:45:00.000Z',
    notes: null,
    entries: [
      {
        exerciseId: 'wger:73',
        exerciseName: 'Bench Press',
        restSeconds: 120,
        notes: null,
        sets: [
          { index: 0, reps: 5, weightKg: 80, completed: true, rpe: null },
          { index: 1, reps: 5, weightKg: 80, completed: false, rpe: null },
        ],
      },
    ],
  };
  return { ...valid, ...overrides };
}

export function entry(doc: unknown = document(), overrides: Partial<WatchInboxEntry> = {}): WatchInboxEntry {
  return {
    id: UUID,
    format: 'kinetiq.watch-workout',
    version: 1,
    payload: typeof doc === 'string' ? doc : JSON.stringify(doc),
    ...overrides,
  };
}
